modulus/utils/graphcast/graph_utils.py

```diff
@@ -74,13 +78,17 @@ def create_heterograph(
         Label of the edge type
     dtype : torch.dtype, optional
         Graph index data type, by default torch.int32
+    num_nodes_dict : dict, optional
+        number of nodes for some node types, see dgl.heterograph for more information
 
     Returns
     -------
     DGLGraph
         The dgl Graph.
     """
-    graph = dgl.heterograph({labels: ("coo", (src, dst))}, idtype=dtype)
+    graph = dgl.heterograph(
+        {labels: ("coo", (src, dst))}, num_nodes_dict=num_nodes_dict, idtype=dtype
```

Comment:
avoid issue with some of the processor graphs at higher resolution erroring if max(idx) in src/dst does not match the specified number of nodes